src/app/call/[callAssId]/layout.tsx

```diff
@@ -4,10 +4,12 @@ import CallLayout from 'features/call/layouts/CallLayout';
 
 type Props = {
   children?: ReactNode;
+  params: { callAssId: string };
 };
 
-const CallPageLayout: FC<Props> = ({ children }) => {
-  return <CallLayout>{children}</CallLayout>;
+const CallPageLayout: FC<Props> = ({ children, params }) => {
+  const { callAssId } = params;
+  return <CallLayout callAssId={callAssId}>{children}</CallLayout>;
```

Comment:
Some logic needs to be moved into this component to make it basically wrap all the routes in the call interface and make sure people are correctly redirected. 

This needs to be in here: 

-  `await redirectIfLoginNeeded();` on the first line, to redirect people to login
- Then all the logic to determine if the user is assigned the call and the org has the feature flag that now lives in `call/[callAssId]/page.tsx` - 404 if they don't have the assignment and redirect to old Gen 2 if org does not have the feature flag 

This means that type of logic can move out from the page.tsx files under /call